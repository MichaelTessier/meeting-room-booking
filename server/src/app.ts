import express from 'express'

import bodyParser from 'body-parser'

import cors from 'cors'
import helmet from 'helmet'
import routes from './api/routes'
import { notFoundMiddleware } from './middlewares/not-found'
import { errorMiddleware } from './middlewares/error'
import { corsMiddleware } from './middlewares/cors'

const app = express()

app.use(express.static('public'))
app.use(bodyParser.json())

app.disable('x-powered-by')
// app.use(
//   cors({
//     origin: 'http://localhost:5173/',
//   }),
// )
app.use(corsMiddleware())
app.use(helmet())

// use body parser
// app.use(express.json())
app.use(express.urlencoded({ extended: true }))

app.use((req, res, next) => {
  res.setHeader('Access-Control-Allow-Origin', '*')
  res.setHeader('Access-Control-Allow-Methods', 'GET, POST')
  res.setHeader('Access-Control-Allow-Headers', 'Content-Type')
  res.setHeader('Access-Control-Allow-Credentials', 'true')
  next()
})

app.use('/api/v1', routes)
app.use(notFoundMiddleware)
app.use(errorMiddleware)

export default app
