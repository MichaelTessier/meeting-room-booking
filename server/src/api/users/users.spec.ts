import { describe, it, expect } from 'vitest'
import request from 'supertest'
import app from '@/app'

describe('API User', () => {
  describe('GET /api/v1/users', () => {
    it('should return a formatted users', async () => {
      const response = await request(app).get('/api/v1/users')
      expect(response.status).toBe(200)
      expect(response.body).toMatchSnapshot()
    })
  })

  describe('GET /api/v1/users/:id', () => {
    it('should return a formatted user', async () => {
      const response = await request(app).get('/api/v1/users/1')
      expect(response.status).toBe(200)
      expect(response.body).toMatchSnapshot()
    })

    it('should return error if unknown id', async () => {
      const response = await request(app).get('/api/v1/users/qsdqs')
      expect(response.status).toBe(404)
    })
  })
})
