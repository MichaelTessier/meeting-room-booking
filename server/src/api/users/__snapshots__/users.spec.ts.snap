// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`API User > GET /api/v1/users > should return a formatted users 1`] = `
[
  {
    "createdAt": "2025-04-07T12:39:29.812Z",
    "id": "1",
    "name": "Iron Man",
    "updatedAt": "2025-04-08T17:31:39.489Z",
  },
  {
    "createdAt": "2025-04-07T12:39:29.812Z",
    "id": "2",
    "name": "Daredevil",
    "updatedAt": "2025-04-08T17:31:39.489Z",
  },
  {
    "createdAt": "2025-04-07T12:39:29.812Z",
    "id": "3",
    "name": "Thor",
    "updatedAt": "2025-04-08T17:31:39.489Z",
  },
]
`;

exports[`API User > GET /api/v1/users/:id > should return a formatted user 1`] = `
{
  "createdAt": "2025-04-07T12:39:29.812Z",
  "id": "1",
  "name": "Iron Man",
  "updatedAt": "2025-04-08T17:31:39.489Z",
}
`;
